Angular/src/app/dialogs/document-filter/document-filter.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { MatDialogRef } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-document-filter',
+  templateUrl: './document-filter.component.html',
+  styleUrls: ['./document-filter.component.scss'],
+})
+export class DocumentFilterComponent {
+  form: any = new FormGroup({
+    author: new FormControl(''),
+    title: new FormControl(''),
+    editor: new FormControl(''),
+    status: new FormControl(''),
+  });
+
+  constructor(public dialogRef: MatDialogRef<DocumentFilterComponent>) {}
+
+  public test() {
+    console.log(this.form.value);
+  }
+
+  protected close_dialog() {
+    const return_form = this.remove_empty_form_fields(this.form.value);
+    this.dialogRef.close(return_form);
+  }
+
+  /**
+   * Removes all form values that are empty from the given object
```

Comment:
Well well, mr. clean

Reply:
Flask would get angry at me for sending authors called `null`.